spectator-api/src/test/java/com/netflix/spectator/impl/SchedulerTest.java

```diff
@@ -88,6 +89,50 @@ public void updateNextFixedRateSkip() {
     Assert.assertEquals(3L, skipped.count());
   }
 
+  private long numberOfThreads(String id) {
+    return Thread.getAllStackTraces()
+        .keySet()
+        .stream()
+        .filter(t -> t.getName().startsWith("spectator-" + id))
+        .count();
+  }
+
+  @Test
+  public void shutdownStopsThreads() throws Exception {
+    Scheduler s = new Scheduler(new DefaultRegistry(), "shutdown", 1);
+
+    // Schedule something to force it to start the threads
+    Scheduler.Options opts = new Scheduler.Options()
+        .withFrequency(Scheduler.Policy.FIXED_RATE_SKIP_IF_LONG, Duration.ofMillis(10))
+        .withStopOnFailure(false);
+    ScheduledFuture<?> f = s.schedule(opts, () -> {});
+    Assert.assertEquals(1L, numberOfThreads("shutdown"));
+
+    // Shutdown and wait a bit, this gives the thread a chance to restart
+    s.shutdown();
```

Comment:
... and then
```
    s.shutdown();
    Assert.assertTrue(latch.await(60, TimeUnit.SECONDS));
    Assert.assertEquals(0L, numberOfThreads("shutdown"));
```

This obviously couples the test to an implementation detail, but it seems likely calling `interrupt()` will remain.